packages/clerk-js/src/core/constants.ts

```diff
@@ -29,3 +29,8 @@ export const SIGN_IN_INITIAL_VALUE_KEYS = ['email_address', 'phone_number', 'use
 export const SIGN_UP_INITIAL_VALUE_KEYS = ['email_address', 'phone_number', 'username', 'first_name', 'last_name'];
 
 export const DEBOUNCE_MS = 350;
+
+export const SIGN_UP_MODES = {
+  PUBLIC: 'public',
+  RESTRICTED: 'restricted',
+} as const;
```

Comment:
Should this include a record type of `Record<string, SignUpModes>` to ensure we don't get out of sync with `SignUpModes` type?